src/hotspot/share/nmt/vmatree.hpp

```diff
@@ -91,15 +91,17 @@ class VMATree {
   private:
     // Store the type and mem_tag as two bytes
     uint8_t type_tag[2];
-    NativeCallStackStorage::StackIndex sidx;
+    NativeCallStackStorage::StackIndex sidx;       // call-stack of all operations
+    NativeCallStackStorage::StackIndex second_idx; // call-stack when committing/uncommitting the start-node of a reserved region
```

Comment:
These can be called `primary_stack` and `secondary_stack`. Does an uncommitted region really have a stack?

Reply:
renamed.

Yes uncommit also has stack although never reported. `VMATree::register_mapping()` will save the stack in the nodes, regardless of the operation. It is done when constructing the `stA` and `stB` interval-states:
```
IntervalChange stA{
      IntervalState{StateType::Released, empty_regiondata},
      IntervalState{              state,   metadata}
  };
  IntervalChange stB{
      IntervalState{              state,   metadata},
      IntervalState{StateType::Released, empty_regiondata}
  };
```
The stack may be empty or whatever passed in by `metadata` arg.